### PR TITLE
Split the API examples into sub-pages

### PR DESCRIPTION
**Pull request by jleibs**
### What
Preview using the latest version of landing (python code samples first, shadows around images, no glitching on scroll, etc.)
 - https://landing-2yxomjvlz-rerun.vercel.app/preview/d3ede5aa8/docs/reference/data_types
  
image

More individual component cleanup will happen in subsequent PRs.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

PR Build Summary: https://build.rerun.io/pr/2438

Docs preview: https://rerun.io/preview/5757435/docs
Examples preview: https://rerun.io/preview/5757435/examples

### CODE REVIEW

File: docs/code-examples/point2d_random.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""Log some random points with color and radii."""
+from __future__ import annotations
```
Comment (nikolausWest) [inline docs/code-examples/point2d_random.py:2]:
```suggestion
```
These only add noise and no value for these examples. Definitely need to go. None of these code examples even contain type annotations

File: docs/code-examples/point3d_random.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+"""Log some random points with color and radii."""
+from __future__ import annotations
```
Comment (nikolausWest) [inline docs/code-examples/point3d_random.py:2]:
```suggestion
```

File: docs/code-examples/point2d_simple.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+"""Log some very simple points."""
+from __future__ import annotations
```
Comment (nikolausWest) [inline docs/code-examples/point2d_simple.py:2]:
```suggestion
```

File: docs/code-examples/point3d_simple.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+"""Log some very simple points."""
+from __future__ import annotations
```
Comment (nikolausWest) [inline docs/code-examples/point3d_simple.py:2]:
```suggestion
```

File: docs/code-examples/point2d_random.py (hunk 1 above)
Comment (abey79) [reply docs/code-examples/point2d_random.py]: This allows to use modern type hints all the way back to 3.8. PyUpgrade (via `just py-format`) will ensure that *if* that thing is there, which is why now ruff enforces it.

Comment (abey79) [reply docs/code-examples/point2d_random.py]: I agree that it's annoying here, though.

Comment (abey79) [reply docs/code-examples/point2d_random.py]: `# isort: skip_file` should disable the rule for the example but we don't what that from showing either. Any means of dropping top-level comments when building the page?

Comment (abey79) [reply docs/code-examples/point2d_random.py]: @jleibs I figured it out. Add the following in `rerun_py/pyproject.toml`, adjusting the path to where those example files are:

```toml
[tool.ruff.per-file-ignores]
"path/to/doc/examples/*" = ["I002"]
```

Comment (jleibs) [reply docs/code-examples/point2d_random.py]: Nice! thanks.

File: docs/content/reference/data_types/point2d.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+---
+title: Point2D
+order: 7
+---
+`Point2d` represents a singular point in two-dimensional space with optional color, radii, and label. `Point2d` entities will be drawn as part of the 2D Spatial SpaceView.
+
+It is compatible with [`AnnotationContext`](../../concepts/annotation-context.md). `class_id` can be used to provide
```
Comment (Wumpf) [inline docs/content/reference/data_types/point2d.md:7]: that's true for most things actually, not sure if we want to repeat that text on all of them? 🤔 
Sadly not all ofc..

File: docs/content/reference/data_types/point2d.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+colors and labels from the annotation context, and `keypoint_id` can be used to make connected edges between points. See
+examples in the `AnnotationContext` documentation.
```
Comment (Wumpf) [inline docs/content/reference/data_types/point2d.md:9]: missing link

**Review by Wumpf — APPROVED** [review]
love the opening page

**Review by abey79 — CHANGES_REQUESTED** [review]
@jleibs I just noticed the spurious `)` in viewer-walkthrough line 240. Can you remove it?

**Review by abey79 — APPROVED** [review]

**Pull request merged**